action.bash

```diff
@@ -33,21 +33,20 @@ push () {
 	fi
 }
 
+re_docker_tag="^refs/(heads|tags)/([^/]+|[a-z]+/(.*))$"
+
 case "$GITHUB_EVENT_NAME" in
-	pull_request)
-		grep -qEe '^refs/pull/[0-9]+' <<<"$GITHUB_REF"
-		TAG="pr$(grep -oEe '[0-9]+' <<<"$GITHUB_REF")"
-		mkimg
-		;;
-	push)
-		grep -qEe '^refs/heads/.' <<<"$GITHUB_REF"
-		TAG="$(cut -d / -f 3- <<<"$GITHUB_REF")"
-		mkimg "$TAG"
-		push
-		;;
+	workflow_dispatch)
+		;&
+	schedule)
+		;&
 	release)
-		grep -qEe '^refs/tags/v[0-9]' <<<"$GITHUB_REF"
-		TAG="$(cut -d v -f 2- <<<"$GITHUB_REF")"
+		[[ "$GITHUB_REF" =~ $re_docker_tag ]]
+		if [[ ${BASH_REMATCH[3]} ]]; then
```

Comment:
As you may have noticed I'm not a big fan of writing $ not in "". Is `[[ ${BASH_REMATCH[3]} ]]` absolutely save? (`$re_docker_tag` is OK for me.)

Reply:
Since git refs won't contain any spaces, it should.

Reply:
In other words, no, the syntax as such is not safe for non-hardcoded values.

Please check this w/o actually **eval**uating the value in the array, e.g. by checking the array's length.

Reply:
Won't work. The array is of length 4 in any case. There's no other option than checking its content.

Reply:
> The array is of length 4 in any case.

OK, but then you have the `[ -n "${...}" ]` option.